java/src/main/java/com/codurance/training/tasks/Execute.java

```diff
@@ -4,43 +4,70 @@
 import java.util.List;
 import java.util.Map;
 
-public class Execute {
-    public void execute(String commandLine, PrintWriter out, Map<String, List<Task>> tasks) {
+public class Execute implements TaskService {
+
+    private final Map<String, List<Task>> tasks;
+    private final PrintWriter out;
+
+    Show show;
+    Add add;
+    Check check ;
+    Uncheck uncheck ;
+    Help help ;
+    Delete delete ;
+    TodayDeadlineTask todayDeadlineTask ;
+    ViewByDeadline viewByDeadline ;
+    ErrorHandler errorHandler;
```

Comment:
type should be of respective implemented interfaces(ex: UpdateTaskService instead of Check ), so that we depend on abstraction than on concrete class